homeassistant/components/fastdotcom/__init__.py

```diff
@@ -42,51 +43,29 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
                 data=config[DOMAIN],
             )
         )
+    await async_setup_services(hass)
     return True
 
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up Fast.com from a config entry."""
     coordinator = FastdotcomDataUpdateCoordindator(hass)
-
-    async def _request_refresh(event: Event) -> None:
-        """Request a refresh."""
-        await coordinator.async_request_refresh()
-
-    async def _request_refresh_service(call: ServiceCall) -> None:
-        """Request a refresh via the service."""
-        ir.async_create_issue(
-            hass,
-            DOMAIN,
-            "service_deprecation",
-            breaks_in_ha_version="2024.7.0",
-            is_fixable=True,
-            is_persistent=False,
-            severity=ir.IssueSeverity.WARNING,
-            translation_key="service_deprecation",
-        )
-        await coordinator.async_request_refresh()
-
-    if hass.state == CoreState.running:
-        await coordinator.async_config_entry_first_refresh()
-    else:
-        # Don't start the speedtest when HA is starting up
-        hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STARTED, _request_refresh)
-
     hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
-    hass.services.async_register(DOMAIN, "speedtest", _request_refresh_service)
 
-    await hass.config_entries.async_forward_entry_setups(
-        entry,
-        PLATFORMS,
-    )
+    async def _async_finish_startup(hass: HomeAssistant) -> None:
+        """Finish setting up the integration."""
+        await coordinator.async_config_entry_first_refresh()
+        await hass.config_entries.async_forward_entry_setups(
+            entry,
+            PLATFORMS,
+        )
 
+    async_at_started(hass, _async_finish_startup)
```

Comment:
Please add a comment why we don't want to run a speedtest when Home Assistant is starting up.